src/olympia/amo/search.py

```diff
@@ -129,57 +128,49 @@ def _build_query(self):
                     source.extend(value)
                 as_list, as_dict = False, True
             elif action == 'query':
-                queries.extend(self._process_queries(value))
+                query &= self._process_queries(value)
             elif action == 'filter':
-                filters.extend(self._process_filters(value))
+                query &= self._process_filters(value)
             elif action == 'source':
                 source.extend(value)
             elif action == 'aggregate':
                 aggregations.update(value)
-            elif action == 'score':
-                functions.append(value)
+            elif action == 'filter_query_string':
+                query_string = value
             else:
                 raise NotImplementedError(action)
 
-        if len(queries) > 1:
-            qs = {'bool': {'must': queries}}
-        elif queries:
-            qs = queries[0]
-        else:
-            qs = {"match_all": {}}
-
-        if functions:
-            qs = {
-                "function_score": {
-                    "query": qs,
-                    "functions": functions
-                }
-            }
-
-        if filters:
-            if len(filters) > 1:
-                filters = {'bool': {'must': filters}}
-
-            qs = {
-                "bool": {
-                    "must": qs,
-                    "filter": filters
-                }
-            }
-
-        body = {"query": qs}
+        # If we have a raw query string we are going to apply all sorts
+        # of boosts and filters to improve relevance scoring.
+        #
+        # We are using the same rules that `search.filters:SearchQueryFilter`
+        # implements to have a single-source of truth for how our
+        # scoring works.
+        from olympia.search.filters import SearchQueryFilter
+
+        query = Search().query(query)
```

Comment:
Potentially fixed in eacf7a6198… I renamed the result of `Search().query()` to `search` which is essentially what it returns, a `Search` object. That means we don't redefine the `query` instance and there's a clearer separation in the `if query_string` statement what the `query_string` is applied to, the actual `search`.

Does that work better?